services/src/main/java/org/keycloak/services/resources/admin/ComponentResource.java

```diff
@@ -289,4 +298,45 @@ private ComponentTypeRepresentation toComponentTypeRepresentation(ProviderFactor
         rep.setMetadata(metadata);
         return rep;
     }
+
+    @GET
+    @Path("/metadata/{factoryId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    @Tag(name = KeycloakOpenAPI.Admin.Tags.COMPONENT)
+    @Operation( summary = "Returns the metadata for a provider factory with the given id.")
+    public ComponentTypeRepresentation getMetadata(@PathParam("factoryId") String factoryId) {
+        auth.realm().requireViewRealm();
+
+        if (factoryId == null) {
+            throw new BadRequestException("Factory id not provided");
+        }
+
+        KeycloakSessionFactory sessionFactory = session.getKeycloakSessionFactory();
+        Set<Spi> spis = sessionFactory.getSpis();
+
+        return spis.stream()
+                .map((Function<Spi, ProviderFactory>) spi -> sessionFactory.getProviderFactory(spi.getProviderClass(), factoryId))
```

Comment:
If we want to have a new method from `KeycloakSessionFactory` to return a provider factory without iterating over all the SPIs, please let me know. If so, we should also update other places like the server info endpoint.